File: java/dsframework/src/main/java/dsf/core/MsgMatcher.java
package dsf.core;

/**
 * 匹配指定名字的消息
 * @author arksea
 */
public class MsgMatcher implements IMessageMatcher {

    private final String name;

    public MsgMatcher(String name) {
        this.name = name;
    }

    @Override
    public Message match(Message msg) throws Throwable {
        return msg.name.equals(name) ? msg : null;
    }
}

File: csharp/dsf/dsf/src/core/MsgMatcher.cs
namespace com.baidu.dsf.core
{

    /// <summary>
    /// 匹配指定名字的消息
    /// @author arksea
    /// </summary>
    public class MsgMatcher : IMessageMatcher
    {

        private readonly string name;

        public MsgMatcher(string name)
        {
            this.name = name;
        }

//JAVA TO C# CONVERTER WARNING: Method 'throws' clauses are not available in .NET:
//ORIGINAL LINE: @Override public Message match(Message msg) throws Throwable
        public virtual Message match(Message msg)
        {
            return msg.name.Equals(name) ? msg : null;
        }
    }

}